apps/finance/app/src/components/Balances.js

```diff
@@ -1,152 +1,170 @@
-const CONVERT_THROTTLE_TIME = 5000
+const CONVERT_API_RETRY_DELAY = 2000
 
-const convertApiUrl = symbols =>
-  `${CONVERT_API_BASE}/price?fsym=USD&tsyms=${symbols.join(',')}`
-
-const keyFromBalances = balances =>
-  balances
-    .filter(({ verified }) => verified)
-    .sort(({ addressA, addressB }) => addressA - addressB)
-    .map(({ address, numData: { amount } }) => `${address}:${amount}`)
-    .join(',')
-
-const areSameBalances = (balancesA, balancesB) => {
-  const keyA = keyFromBalances(balancesA)
-  const keyB = keyFromBalances(balancesB)
-  return keyA === keyB
+function convertRatesUrl(symbolsQuery) {
+  return `https://min-api.cryptocompare.com/data/price?fsym=USD&tsyms=${symbolsQuery}`
 }
 
-class Balances extends React.Component {
-  state = {
-    convertRates: {},
-  }
-  componentDidMount() {
-    this.updateConvertedRates(this.props)
-  }
-  componentDidUpdate(prevProps) {
-    if (!areSameBalances(prevProps.balances, this.props.balances)) {
-      this.updateConvertedRates(this.props)
-    }
-  }
-  updateConvertedRates = throttle(async ({ balances }) => {
-    const verifiedSymbols = balances
-      .filter(({ verified }) => verified)
-      .map(({ symbol }) => symbol)
-
-    if (!verifiedSymbols.length) {
-      return
-    }
+function useConvertRates(symbols) {
+  const [rates, setRates] = useState({})
 
-    const res = await fetch(convertApiUrl(verifiedSymbols))
-    const convertRates = await res.json()
-    this.setState({ convertRates })
-  }, CONVERT_THROTTLE_TIME)
-
-  render() {
-    const { compactMode, balances, theme } = this.props
-    const { convertRates } = this.state
-    const balanceItems = balances.map(
-      ({ address, numData: { amount, decimals }, symbol, verified }) => {
-        const adjustedAmount = amount / Math.pow(10, decimals)
-        const convertedAmount =
-          verified && convertRates[symbol]
-            ? adjustedAmount / convertRates[symbol]
-            : -1
-        return {
-          address,
-          symbol,
-          verified,
-          amount: round(adjustedAmount, 5),
-          convertedAmount: round(convertedAmount, 5),
+  const symbolsQuery = symbols.join(',')
+
+  useEffect(() => {
+    let cancelled = false
+    let retryTimer = null
+
+    const update = async () => {
+      if (!symbolsQuery) {
+        setRates({})
+        return
+      }
+
+      try {
+        const response = await fetch(convertRatesUrl(symbolsQuery))
+        const rates = await response.json()
+        if (!cancelled) {
+          setRates(rates)
+        }
+      } catch (err) {
+        // The !cancelled check is needed in case:
+        //  1. The fetch() request is ongoing.
+        //  2. The component gets unmounted.
+        //  3. An error gets thrown.
+        //
+        //  Assuming the fetch() request keeps throwing, it would create new
+        //  requests even though the useEffect() got cancelled.
+        if (!cancelled) {
```

Comment:
I think it could be important to keep trying, e.g. if the service is temporarily down. What do you think of adding more delay one every attempt? https://github.com/aragon/aragon-apps/pull/1121/commits/2ebcf8978d41b2d6d62fe0f50ed830c8e9431822